### ISSUE TITLE
Curl Error while trying to setup tomcat

### ISSUE THREAD

**Issue by kblodget** [antoineco/aco-tomcat#76]
I am a new user of Puppet so please forgive me if I have overlooked something simple.
I am using Vagrant 1.9.5 and Puppet 4.3.2 to create tomcat
I have a manifest file configured  to start a new instance of tomcat
```puppet
class someVariable::tomcat ( 
  $tomcat_version = '8.5.15'
) {

  class { '::tomcat':
    install_from    => 'archive',
    version         => "${tomcat_version}",
    log4j           => true,
    log4j_enable    => true
  }
}
```

When I run this in Vagrant I get the following error:
```
Notice: /Stage[main]/Tomcat::Log4j/File[global log4j library]/ensure: created
Notice: /Stage[main]/Tomcat::Log4j/File[global log4j ini configuration]/ensure: defined content as '{md5}5ceb73fffe253461f415afd1a5c853f2'
Notice: /Stage[main]/Tomcat::Log4j/File[global logging configuration]/ensure: removed
Notice: /Stage[main]/Tomcat::Extras/File[global extras directory]/ensure: created
Notice: /Stage[main]/Tomcat::Extras/Archive[catalina-jmx-remote.jar]/ensure: download archive from http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/catalina-jmx-remote.jar to /usr/share/tomcat8/lib/extras/catalina-jmx-remote-8.5.15.jar  without cleanup
Notice: /Stage[main]/Tomcat::Extras/Archive[catalina-ws.jar]/ensure: download archive from http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/catalina-ws.jar to /usr/share/tomcat8/lib/extras/catalina-ws-8.5.15.jar  without cleanup
Error: Execution of '/bin/curl http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/tomcat-juli-adapters.jar -o /tmp/tomcat-juli-adapters-8.5.15.jar_20170607-4208-nabgdk -fsSL --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Error: /Stage[main]/Tomcat::Extras/Archive[tomcat-juli-adapters.jar]/ensure: change from absent to present failed: Execution of '/bin/curl http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/tomcat-juli-adapters.jar -o /tmp/tomcat-juli-adapters-8.5.15.jar_20170607-4208-nabgdk -fsSL --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Error: Execution of '/bin/curl http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/tomcat-juli.jar -o /tmp/tomcat-juli-extras-8.5.15.jar_20170607-4208-v3b3tn -fsSL --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Error: /Stage[main]/Tomcat::Extras/Archive[tomcat-juli-extras.jar]/ensure: change from absent to present failed: Execution of '/bin/curl http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/tomcat-juli.jar -o /tmp/tomcat-juli-extras-8.5.15.jar_20170607-4208-v3b3tn -fsSL --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Notice: /Stage[main]/Tomcat::Extras/File[catalina-jmx-remote.jar]/ensure: created
Notice: /Stage[main]/Tomcat::Extras/File[catalina-ws.jar]/ensure: created
Notice: /Stage[main]/Tomcat::Extras/File[tomcat-juli-adapters.jar]/ensure: created
Notice: /Stage[main]/Tomcat::Extras/File[tomcat-juli-extras.jar]/ensure: created
Notice: /Stage[main]/Tomcat::Extras/File[tomcat-juli.jar]/ensure: ensure changed 'file' to 'link'
```
I need to wrap the call to create the instance of tomcat because I have to setup SSL later in the manifest. 

It appears the curl command is trying to access 
``
http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/tomcat-juli-adapters.jar
``
which doesn't exist in this distribution. How can I get past this error to create a tomcat server

**Comment by antoineco** [comment]
Nice catch :) It seems like the `tomcat-juli` extra library is missing for this version: http://archive.apache.org/dist/tomcat/tomcat-8/v8.5.15/bin/extras/

8.5.3 is actually the last version for which this lib is available. I'll investigate why.

In the meantime you can either use tomcat 8.5.3 or use the standard logger instead of log4j (your apps will still use log4j, only the internal tomcat logging won't).

Thanks for reporting.

**Comment by antoineco** [comment]
Ok, found it:

> 58588: Remove the JULI extras package from the distribution. It was only useful for switching Tomcat's internal logging to log4j 1.2.x and that version of log4j is no longer supported. No additional Tomcat code is required if switching Tomcat's internal logging to log via log4j 2.x. (markt)

from: https://tomcat.apache.org/tomcat-8.5-doc/changelog.html

Looks like I just have to omit that library for tomcat > 8.5.3.

**Comment by kblodget** [comment]
Thank you for the quick response. I really would prefer to use your module over puppetlabs-tomcat so hopefully this will be a quick fix.

**Comment by antoineco** [comment]
@kblodget unfortunately few Linux distributions have a package for log4j 2.0, so there are 2 options:

* get the latest binary distribution from apache using this module
* drop support for log4j

A while ago I've been considering dropping that option altogether because log4j is used very seldom by tomcat itself. Do you use it yourself for tomcat internal logging?

**Comment by kblodget** [comment]
I can work without log4j support

**Comment by antoineco** [comment]
Released in v1.8.1.
